services/worker/src/worker/statistics_utils.py

```diff
@@ -456,6 +477,24 @@ def is_class(n_unique: int, n_samples: int) -> bool:
             n_unique / n_samples <= MAX_PROPORTION_STRING_LABELS and n_unique <= MAX_NUM_STRING_LABELS
         ) or n_unique <= NUM_BINS
 
+    @staticmethod
+    def is_datetime(data: pl.DataFrame, column_name: str) -> tuple[bool, Optional[str]]:
+        """Check if first 1000 non-null samples in a column match datetime format. If true, also return datetime format"""
+
+        values = data.filter(pl.col(column_name).is_not_null()).head(1000)[column_name].to_list()
```

Comment:
you can lower to first 100 to reduce the risk of OOM imo

```suggestion
        """Check if first 100 non-null samples in a column match datetime format. If true, also return datetime format"""

        values = data.filter(pl.col(column_name).is_not_null()).head(100)[column_name].to_list()
```

Reply:
do you think there might be OOM with 1000 texts?

Reply:
some datasets have 1 row = 1 full book so it can oom in rare cases like that